Sources/TensorFlow/Layers/Dropout.swift

```diff
@@ -68,24 +73,28 @@ public struct Dropout<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
 ///
 /// The noise added always has mean zero, but has a configurable standard deviation.
 public struct GaussianNoise<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
+  /// The input type of the layer.
+  public typealias Input = Tensor<Scalar>
+  /// The output type of the layer.
+  public typealias Output = Tensor<Scalar>
   public typealias TangentVector = EmptyTangentVector
 
-  @noDerivative public let standardDeviation: Tensor<Scalar>
+  @noDerivative public let standardDeviation: Input
```

Comment:
With changes to the `Module` and `Layer` protocols in this PR, it seems that conforming types now need to explicitly specify `Input` and `Output` typealiases. (Previously, the compiler could infer `Input` and `Output` associated types for these conforming types, but that doesn't seem to work anymore.)

Adding `Input` and `Output` typealiases is fine, but how about keeping `Tensor<Scalar>` occurrences explicit instead of renaming them to `Input` and `Output`?
* `Input` and `Output` are closely tied to the `func callAsFunction(_: Input) -> Output` protocol requirement, so it's a bit strange to use them outside of `func callAsFunction`, like as the type of stored properties (which are neither layer inputs nor outputs).
* `Tensor<Scalar>` (a specific concrete type) also conveys more information than `Input` and `Output` (associated types), so it's easier to understand usages like `Tensor<Scalar>(randomNormal: ...)` than `Input(randomNormal: ...)`.

Reply:
That makes sense, I'll revert those changes everywhere outside `callAsFunction()`.

Reply:
Thank you!

Reply:
I was actually able to remove these typealiases altogether, which makes this PR much smaller 👍

Reply:
Yay, even better!